WordPressVIPMinimum/Sniffs/Compatibility/RestrictedCacheGroupSniff.php

```diff
@@ -0,0 +1,134 @@
+
+	/**
+	 * Functions that can assign or modify cache key.
+	 *
+	 * @var array <string function name> => <bool (true)>
+	 */
+	protected $target_functions = [
+		'wp_cache_add'     => true,
+		'wp_cache_set'     => true,
+		'wp_cache_replace' => true,
+	];
+
+	/**
+	 * List of cache group names already in use by wp-memcached.
+	 *
+	 * @link https://github.com/Automattic/wp-memcached/blob/master/readme.txt
+	 * @var array
+	 */
+	private $wp_memcached_groups = [
+		'blog-details'              => true,
+		'blog-id-cache'             => true,
+		'blog-lookup'               => true,
+		'bookmark'                  => true,
+		'calendar'                  => true,
+		'category'                  => true,
+		'category_relationships'    => true,
+		'comment'                   => true,
+		'counts'                    => true,
+		'general'                   => true,
+		'global-posts'              => true,
+		'options'                   => true,
+		'plugins'                   => true,
+		'post_ancestors'            => true,
+		'post_format_relationships' => true,
+		'post_meta'                 => true,
+		'post_tag_relationships'    => true,
+		'posts'                     => true,
+		'rss'                       => true,
+		'site-lookup'               => true,
+		'site-options'              => true,
+		'site-transient'            => true,
+		'term_meta'                 => true,
+		'terms'                     => true,
+		'themes'                    => true,
+		'timeinfo'                  => true,
+		'transient'                 => true,
+		'user_meta'                 => true,
+		'useremail'                 => true,
+		'userlogins'                => true,
+		'usermeta'                  => true,
+		'users'                     => true,
+		'userslugs'                 => true,
+		'widget'                    => true,
+
+	];
+
+	/**
+	 * Process the parameters of a matched function.
+	 *
+	 * @param int    $stackPtr        The position of the current token in the stack.
+	 * @param array  $group_name      The name of the group which was matched.
+	 * @param string $matched_content The token content (function name) which was matched.
+	 * @param array  $parameters      Array with information about the parameters.
+	 * @return int|void Integer stack pointer to skip forward or void to continue
+	 *                  normal file processing.
+	 */
+	public function process_parameters( $stackPtr, $group_name, $matched_content, $parameters ) {
+		if ( ! isset( $parameters[3] ) ) {
+			return; // Bail, less than 3 parameters.
+		}
+
+		$scope_end = $parameters[3]['end'] + 1;
+
+		$group = $this->phpcsFile->findNext(
+			Tokens::$textStringTokens,
+			$parameters[3]['start'],
+			$scope_end
+		);
+
+		if ( $group === false ) {
+			return;
+		}
+
+		$is_concat = $this->phpcsFile->findNext(
+			Tokens::$emptyTokens,
+			$group + 1,
+			$scope_end,
+			true
+		);
+
+		if ( $this->tokens[ $is_concat ]['code'] === T_STRING_CONCAT ) {
+			return; // Bail, indeterminable.
+		}
```

Comment:
What are you trying to do here ? Or rather: what _should_ the sniff do here ?

Please re-read my previous comment about the logic needed. This feels like you are trying to take a shortcut instead of _walking the tokens_, which is what is needed.

Think:
```php
wp_cache_set( $key, [ 1, 2, 3 ],  $prefix . 'post_meta' );
wp_cache_set( $key, [ 1, 2, 3 ],  'post_meta' + $nr );
```